nmmo/core/observation.py

```diff
@@ -24,30 +26,46 @@ def __init__(self,
 
     entities = entities[0:config.PLAYER_N_OBS]
     entity_ids = entities[:,EntityState.State.attr_name_to_col["id"]]
+    entity_pos = entities[:,[EntityState.State.attr_name_to_col["row"],
+                             EntityState.State.attr_name_to_col["col"]]]
     self.entities = SimpleNamespace(
         values = entities,
         ids = entity_ids,
-        id = lambda i: entity_ids[i] if i < len(entity_ids) else None
+        len = len(entity_ids),
+        id = lambda i: entity_ids[i] if i < len(entity_ids) else None,
+        index = lambda val: np.nonzero(entity_ids == val)[0][0] if val in entity_ids else None,
+        pos = entity_pos,
+        # for the distance function, see io/action.py, Attack.call(), line 222
+        dist = lambda pos: np.max(np.abs(entity_pos - np.array(pos)), axis=1),
```

Comment:
this seems wrong, you need to sum the abs distances, not max them. Do you need a dist function at all, it might be better to have within_range = lambda pos, range: ....

Reply:
now using the l-inf distance function from `nmmo.lib.utils`